client/driver/docker.go

```diff
@@ -153,8 +153,14 @@ func createContainer(ctx *ExecContext, task *structs.Task, logger *log.Logger) d
 		hostConfig.PortBindings = dockerPorts
 	}
 
+	// Merge Nomad-native with user-specified envvars
+	env := TaskEnvironmentVariables(ctx, task).List()
+	for k, v := range task.Env {
+		env = append(env, fmt.Sprintf("%s=%s", k, v))
```

Comment:
I don't think we want to do that as we want to set the environment variables the user passes exactly as they define them. Capital environment variables are only convention.